libarchive/archive_read_disk_posix.c

```diff
@@ -109,6 +109,11 @@ __FBSDID("$FreeBSD$");
 #define O_CLOEXEC	0
 #endif
 
+#if defined(__hpux)
```

Comment:
I would like this better:
```c
#if defined(__hpux) && !defined(HAVE_DIRFD)
```

Reply:
Makes sense as a safety net, will change

Reply:
Done